core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -44,7 +45,9 @@ object RequestChannel extends Logging {
     byteBuffer
   }
 
-  case class Request(processor: Int, connectionId: String, private var buffer: ByteBuffer, startTimeMs: Long, securityProtocol: SecurityProtocol) {
+  case class Session(principal: Principal, host: String)
```

Comment:
``Session`` would have an ``id: Int`` to identify it, right? I mean, in the context of various "anonymous" principal connecting to the same host we need a way of uniquely identifying each session.